tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,875 @@
+                    break
+    
+            (dirname, filename, name, extension) = self.split_name(dirName)
+            # Do not include recordings in diff
+            if (filename == "recorder" and dirname.endswith(
+                "/{}".format(sname))):
+                subdirList[:] = []
+                continue
+
+            if (is_leveldb_rocksdb_dir):
+                logger.debug("{} is a LevelDB/RocksDB directory!"
+                    .format(dirName))
+
+                # Use "nodename" in place of the node's name when a directory
+                # (dirName in this case) matches the node name (sname)
+                if (sname == filename):
+                    filename = "nodename"
+
+                if (filename.endswith("_transactions")):
+                    logger.debug("{} is a transaction LevelDB/RocksDB instance!"
+                        .format(filename))
+                    # utf-8 - call "getAllTxn()" in ledger.py in the ledger
+                    #         module/package in indy-plenum.
+                    # Create a Ledger domain object from the ledger directory 
+                    # 'filename' takes the form {fileNamePrefix}_transactions
+                    # where fileNamePrefix is one of the following:
+                    #    config, pool, domain
+                    # FileHashStore requires a fileNamePrefix.
+                    fileNamePrefix = filename.split("_")[0]
+                    ledger = Ledger(
+                        CompactMerkleTree(
+                            # The FileHashStore reads
+                            # {fileNamePrefix}_merkleLeaves and
+                            # {fileNamePrefix}_merkleNodes from dataDir
+                            hashStore=FileHashStore(
+                                dataDir=dirname,
+                                fileNamePrefix=fileNamePrefix
+                            )
+                        ),
+                        dataDir=dirname,
+                        fileName=filename
+                    )
+
+                    out_filename = join(sout, filename)
+                    with open(out_filename, "a") as file_handle:
+                        for txn in ledger.getAllTxn():
+                            logger.info("Writing >txn={}< to {}".format(
+                                str(txn), out_filename))
+                            file_handle.write(str(txn))
+                    ledger.stop()
+                else:
+                    debug_msg = ("{} is NOT a transaction LevelDB/RocksDB"
+                                 "instance!")
+                    logger.debug(debug_msg.format(filename))
+                    # Assume utf-8 decoding by default.
+                    # utf-8 - use standard leveldb/rocksdb iterator() and utf-8
+                    #         decoding.
+                    opened = False
+                    try:
+                        debug_msg = ("Opening {} as a LevelDB instance with"
```

Comment:
FROM ALEX:
Please note that KeyValueStorageRocksdbIntKeys is used for the ledger